Source/Eto.WinForms/WinConversions.cs

```diff
@@ -403,7 +403,7 @@ public static MouseButtons ToEto(this swf.MouseButtons button)
 
 		public static Graphics ToEto(this sd.Graphics g)
 		{
-			return new Graphics(new GraphicsHandler(g));
+			return new Graphics(new GraphicsHandler(g, false));
```

Comment:
This is **problem of ownership**, you are converting one disposable to another disposable, but the fact you have disposable object at your hand does not automatically mean you have the ownership and should dispose it. Maybe both `ToEto` and `GraphicsHandler` constructor should have non-default bool parametr to let you state it clearly (wether you own the disposable object and want it destroyed or not).

Actually, the very `OnPaint` of the handler could be written with `using(var g = new Graphics(...))` to dispose what you create (**it is disposable after all, it should be disposed by the creator!** what if the object holds some unmanaged resources to provide new features?!), but you do not own the `sd.Graphics` and should therefore **not pass ownership**.

Another question is: will eto Graphics dispose the original when and only when Dispose is called or will it do the same in destructor (called by garbage collector)? Is the "native" graphics treated as managed or non-managed resource? Luckily, `WidgetHandler` treats it as managed (which is correct) and won't dispose it in destructor, luckily, the code was not a disaster as it looked to be :D

...this time, but what next time?
...it is up to you, but I would prefer non-default bool argument to force anobody to think twice, before using the conversion.